### PR TITLE
Join a match

### PR DESCRIPTION
**Pull request by rsun45**

### CODE REVIEW

File: server/models/UserModel.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+const mongoose = require('mongoose');
```
Comment (sina-jamshidi) [inline server/models/UserModel.js]: wait this model already exists though right?

File: server/models/MatchModel.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
```
Comment (sina-jamshidi) [inline server/models/MatchModel.js]: You can discuss this in your standup if it's outside of the scope of your task, but there could probably be more useful information added to this model

File: server/controllers/match.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  Match.findOneAndUpdate(
+    {_id: matchID},
+    {$push: {'userIDs': {id}}},
+    {new: true}
+  ).then((meg) => {
+    console.log(meg);
+  })
+  .catch((err) => {
+    console.log(err);
+  });
```
Comment (sina-jamshidi) [inline server/controllers/match.js]: do we not want to return anything here?

File: server/controllers/match.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+// join a match by given matchID
+const joinMatch = (req, res) => {
+  const matchID = req.params;
+  const id = req.userID;
+  Match.findOneAndUpdate(
+    {_id: matchID},
+    {$push: {'userIDs': {id}}},
+    {new: true}
+  ).then((meg) => {
```
Comment (sina-jamshidi) [inline server/controllers/match.js]: `message`?

**Review by sina-jamshidi — APPROVED** [review]
Make sure to merge in the latest dev to resolve conflicts, also since there's overlapping work between this PR and the other one make sure the latest changes are reflected here 👍 
Looks good to merge once comments are addressed and if it accomplishes what Jess is looking for

File: server/models/UserModel.js (hunk 1 above)
Comment (rsun45) [reply server/models/UserModel.js]: Yes, I merged Umer's branch to this branch for code integrality. They are the same model, no change was made on it.

File: server/controllers/match.js (hunk 3 above)
Comment (rsun45) [reply server/controllers/match.js]: Changed to send status Json when insert player successfully.

File: server/controllers/match.js (hunk 4 above)
Comment (rsun45) [reply server/controllers/match.js]: Modified.

File: server/controllers/match.js [hunk 5]
```diff
@@ -18,11 +18,11 @@ const joinMatch = (req, res) => {
     {_id: matchID},
     {$push: {'userIDs': {id}}},
     {new: true}
-  ).then((meg) => {
-    console.log(meg);
+  ).then(() => {
+    res.status(200).jason({status: 'Joining match'});
```
Comment (uzher-code) [inline server/controllers/match.js]: jason should be changed to json here.

**Review by uzher-code — CHANGES_REQUESTED** [review]

File: server/controllers/match.js (hunk 5 above)
Comment (uzher-code) [reply server/controllers/match.js]: a good way to test if your backend is working properly and sending the correct response is by using a tool like postman

Comment (rsun45) [reply server/controllers/match.js]: Error is modified.

**Pull request merged**